src/main/java/frc/robot/subsystems/ElevatorSubsystem.java

```diff
@@ -0,0 +1,65 @@
+package frc.robot.subsystems;
+
+import com.ctre.phoenix.motorcontrol.ControlMode;
+import com.ctre.phoenix.motorcontrol.can.TalonFX;
+import com.revrobotics.CANSparkMax;
+
+import edu.wpi.first.math.controller.ElevatorFeedforward;
+import edu.wpi.first.networktables.NetworkTable;
+import edu.wpi.first.networktables.NetworkTableEntry;
+import edu.wpi.first.networktables.NetworkTableInstance;
+import edu.wpi.first.wpilibj.DigitalInput;
+import edu.wpi.first.wpilibj2.command.SubsystemBase;
+
+public class ElevatorSubsystem extends SubsystemBase {
+
+    NetworkTableInstance nt = NetworkTableInstance.getDefault();
+    NetworkTable ntArm = nt.getTable("/components/arm");
+    NetworkTableEntry ntArmAngle = nt.getEntry("pivotPosition");
+    NetworkTableEntry ntArmLength = nt.getEntry("telescopeLength");
+
+
+    DigitalInput topLimitSwitch = new DigitalInput(3);
+    DigitalInput bottomLimitSwitch = new DigitalInput(4);
+    
+    private TalonFX elevatorMotor;
+
+
+    public ElevatorSubsystem(TalonFX elevatorMotor) {
+        this.elevatorMotor = elevatorMotor;
+
+        this.elevatorMotor.selectProfileSlot(0, 0);
+        this.elevatorMotor.config_kP(0, 0);
+        this.elevatorMotor.config_kI(0, 0);
+        this.elevatorMotor.config_kD(0, 0);
+        this.elevatorMotor.config_kF(0, 0);
+    }
+
+    public void setElevatorMotor(double speed) {
+        if (bottomLimitSwitch.get() && speed < 0)
```

Comment:
No hardwire limit switch? The talon fx supports it

Reply:
Asked electricians, there was no reason to plug limit switch into roborio instead of talonfx, but it doesn't seem worth it to change now.